multiview/cluster/mv_k_means.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Implements multi-view kmeans clustering algorithm for data with 2-views.
+
+
+import numpy as np
+from .base_cluster import BaseCluster
+from ..utils.utils import check_Xs
+from sklearn.exceptions import NotFittedError, ConvergenceWarning
+
+
+class MultiviewKMeans(BaseCluster):
+
+    '''
+    An implementation of Multi-View K-Means using the co-EM algorithm.
+    This algorithm currently handles two views of data.
+
+    Paramters
+    ---------
+    n_clusters : int (default=2)
+        The number of clusters
+
+    random_state : int (default=None)
+        Determines random number generation for initializing centroids.
+        Can seed the random number generator with an int.
+
+    patience : int, optional (default=5)
+        The number of EM iterations with no decrease in the objective
+        function after which the algorithm will terminate.
+
+    max_iter : int, optional (default=None)
+        The maximum number of EM iterations to run before
+        termination.
+
+    Attributes
+    ----------
+
+    _n_clusters : int
+        The number of clusters
+
+    _random_state : int
+        The seed for the random number generator used during centroid
+        initialization.
+
+    _centroids : list of array_likes
+        - _centroids shape: (2,)
+        - _centroids[0] shape: (n_clusters, n_features_i)
+        The cluster centroids for each of the two views. _centroids[0]
+        corresponds to the centroids of view 1 and _centroids[1] corresponds
+        to the centroids of view 2.
+
+    _patience : int
+        The number of EM iterations with no decrease in the objective
+        function after which the algorithm will terminate.
+
+    _max_iter : int
+        The maximum number of EM iterations to run before
+        termination.
+
+    References
+    ----------
+    [1] Bickel S, Scheffer T (2004) Multi-view clustering. Proceedings of the
+    4th IEEE International Conference on Data Mining, pp. 19–26
+    '''
+
+    def __init__(self, n_clusters=2, random_state=None,
+                 patience=5, max_iter=None):
+
+        super().__init__()
+
+        if not (isinstance(n_clusters, int) and n_clusters > 0):
```

Comment:
note that in the sklearn API typechecking is done on fit, not init. In graspy we did it this way also, however, but if you actually want to be super compliant then you should move this stuff

I think the only thing it actually matters for in practice is with some of the pipeline stuff they have in sklearn. Have a look at the link i sent in slack and decide what you all want to do as a group